src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java

```diff
@@ -1,31 +1,140 @@
 package com.endava.internship.mocking.service;
 
+import com.endava.internship.mocking.model.Payment;
+import com.endava.internship.mocking.model.Status;
+import com.endava.internship.mocking.model.User;
+import com.endava.internship.mocking.repository.InMemUserRepository;
+import com.endava.internship.mocking.repository.PaymentRepository;
+import com.endava.internship.mocking.repository.UserRepository;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
 import org.mockito.junit.jupiter.MockitoExtension;
 
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
 @ExtendWith(MockitoExtension.class)
 class PaymentServiceTest {
 
+    private UserRepository userRepository;
+
+    private PaymentRepository paymentRepository;
+
+    private ValidationService validationService;
+
+    private PaymentService paymentService;
 
     @BeforeEach
     void setUp() {
+        userRepository = mock(UserRepository.class);
+        paymentRepository = mock(PaymentRepository.class);
+        validationService = mock(ValidationService.class);
+        paymentService = new PaymentService(userRepository,
+                paymentRepository, validationService);
+    }
+
+    @Test
+    void createPaymentWhenUserIdNotContained() {
+        Integer userId = 10;
+        Double amount = 2D;
+
+        assertThrows(NoSuchElementException.class,
+                () -> paymentService.createPayment(userId, amount));
+        verify(validationService).validateUserId(userId);
+        verify(validationService).validateAmount(amount);
+        verify(userRepository).findById(userId);
+    }
+
+    @Test
+    void createPaymentWhenUserIsContained() {
+        InMemUserRepository inMemUserRepository = spy(InMemUserRepository.class);
+        paymentService = new PaymentService(inMemUserRepository,
+                paymentRepository, validationService);
+        ArgumentCaptor<Payment> paymentCaptor = ArgumentCaptor.forClass(Payment.class);
+
```

Comment:
use when (userRepository,findById(1).thenReturn(user);
same for payement repository